crates/primitives/src/block.rs

```diff
@@ -222,6 +341,156 @@ pub struct SealedBlock {
+                eip_4844_active in any::<bool>(),
+                blob_gas_used in any::<u64>(),
+                excess_blob_gas in any::<u64>(),
+                parent_beacon_block_root in any::<B256>()
+            ) -> Header {
+                // EIP-1559 logic
+                if header.base_fee_per_gas.is_none() {
+                    // If EIP-1559 is not active, clear related fields
+                    header.withdrawals_root = None;
+                    header.blob_gas_used = None;
+                    header.excess_blob_gas = None;
+                    header.parent_beacon_block_root = None;
+                } else {
+                    // 4895 is not active
+                    if header.withdrawals_root.is_none() {
+                        header.blob_gas_used = None;
+                        header.excess_blob_gas = None;
+                        header.parent_beacon_block_root = None;
+                    } else {
+                        // 4895 is active
+                        if eip_4844_active {
+                            // EIP-4844 is active
+                            header.blob_gas_used = Some(blob_gas_used);
+                            header.excess_blob_gas = Some(excess_blob_gas);
+                            header.parent_beacon_block_root = Some(parent_beacon_block_root);
+                        } else {
+                            // EIP-4844 is not active
+                            header.blob_gas_used = None;
+                            header.excess_blob_gas = None;
+                            header.parent_beacon_block_root = None;
+                        };
+                    }
+                }
+
+                header
+            }
+        }
+        let header_strategy = {
+            <Self as Default>::default();
+            SealedHeader::arbitrary_with(())
+        };
+        let body_strategy = vec(any::<TransactionSigned>(), 0..10);
+        let ommers_strategy = vec(valid_header_strategy(), 0..2);
+        let withdrawals_strategy = any::<Option<Withdrawals>>();
+
+        (header_strategy, body_strategy, ommers_strategy, withdrawals_strategy)
+            .prop_map(|(header, body, ommers, withdrawals)| Self {
+                header,
+                body,
+                ommers,
+                withdrawals,
+            })
+            .boxed()
+    }
+}
+
+#[cfg(any(test, feature = "arbitrary"))]
+impl<'a> arbitrary::Arbitrary<'a> for SealedBlock {
+    fn arbitrary(u: &mut arbitrary::Unstructured<'a>) -> arbitrary::Result<Self> {
+        let mut header: Header = Header::arbitrary(u)?;
+        let blob_gas_used: u64 = u.arbitrary()?;
+        let excess_blob_gas: u64 = u.arbitrary()?;
+        let parent_beacon_block_root: B256 = u.arbitrary()?;
+        // Determine if EIP-4844 is active using a random bool
+        let eip_4844_active: bool = u.arbitrary()?;
+        // EIP-1559 logic
+        #[allow(clippy::collapsible_else_if)]
```

Comment:
remove this, if clippy says it's collapsible, the clippy suggestion should be followed